### PR TITLE
Add recipe for duecredit

### PR DESCRIPTION
**Pull request by mscheltienne**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Closes https://github.com/conda-forge/staged-recipes/issues/4303

@yarikoptic @smoia Please respond in a comment to this PR that you are willing to maintain this feedstock. I also added myself, to enable the auto-merge bot on the created feedstock and to help if needed. Since this is a noarch build, it should be easy to maintain.

### CODE REVIEW

**Comment by mscheltienne** [comment]
@conda-forge/help-python Hello, could someone give me a hint on this failure? I never seen that one yet.

```
duecredit 0.9.1 has requirement citeproc-py>=0.4, but you have citeproc-py a4d73f8-dirty.
```

With installed version `citeproc-py:        0.6.0-py311h38be061_2     conda-forge`. Broken Linux version of `citeproc-py`?

**Comment by smoia** [comment]
Hey there!
I did not work on duecredit that much. I can act as a last-resort maintainer due to my packages depending on duecredit, but @yarikoptic, @mvdoc, @emirvine, @jwodder, and @tsalo might be more indicated for the task?

**Comment by mscheltienne** [comment]
Sure, note that this is not to maintain `duecredit` itself, but the feedstock (the auto-release on `conda-forge`).

**Comment by yarikoptic** [comment]
> With installed version `citeproc-py: 0.6.0-py311h38be061_2 conda-forge`. Broken Linux version of `citeproc-py`?

it is due to a ad-hoc distribution/versioning setup in citeproc, proposing patch against it which I believe should work for conda-forge: https://github.com/brechtm/citeproc-py/pull/131 . Proposed as patch for feedstock: https://github.com/conda-forge/citeproc-py-feedstock/pull/4

**Comment by mscheltienne** [comment]
@yarikoptic All green now, thanks. I am not familiar with the legacy packaging/distribution.
Only point missing before creating the feedstock is a comment from the listed maintainers to acknowledge that they are willing to maintain the feedstock.

I can give a hand, and I am willing to maintain the feedstock along other people. @yarikoptic Let me know if that works for you, and who you would like to have in the list of maintainers in the recipe.

File: recipes/duecredit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/duecredit/duecredit
+  summary: Automated collection and reporting of citations for used software/methods/datasets
+  license: BSD-3-Clause
+  license_family: MIT
```
Comment (dopplershift) [inline recipes/duecredit/meta.yaml:44]:
```suggestion
  license_family: BSD
```

File: recipes/duecredit/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  number: 0
+  entry_points:
+    - duecredit = duecredit.cmdline.main:main
+
+requirements:
+  host:
+    - pip
+    - python >=3.8
+    - setuptools >=61.0.0
+    - wheel
```
Comment (dopplershift) [inline recipes/duecredit/meta.yaml:24]: Is this actually explicitly required to run `setup.py`? It didn't look like it.
```suggestion
```

File: recipes/duecredit/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - duecredit
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/duecredit/duecredit
+  summary: Automated collection and reporting of citations for used software/methods/datasets
+  license: BSD-3-Clause
```
Comment (dopplershift) [inline recipes/duecredit/meta.yaml:43]:
```suggestion
  license: BSD-2-Clause
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just a few minor tweaks, mainly for license metadata. Then this is good to go.

File: recipes/duecredit/meta.yaml (hunk 2 above)
Comment (mscheltienne) [reply recipes/duecredit/meta.yaml:24]: Probably not, I'm not sure how `duecredit` is packaged; this is part of a template I'm using.

**Pull request merged**